src/fairchem/core/modules/element_references.py

```diff
@@ -0,0 +1,206 @@
+"""
+Copyright (c) Meta, Inc. and its affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+from __future__ import annotations
+
+import logging
+from functools import partial
+from typing import TYPE_CHECKING
+
+import numpy as np
+import torch
+from torch import nn
+from torch.utils.data import DataLoader, Dataset
+from tqdm import tqdm
+
+from fairchem.core.datasets import data_list_collater
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from torch_geometric.data import Batch
+
+
+class LinearReference(nn.Module):
+    """Compute a linear reference for target scalar properties"""
+
+    def __init__(
+        self,
+        element_references: torch.Tensor | None = None,
+        max_num_elements: int = 118,
+    ):
+        """
+        Args:
+            element_references (Tensor): tensor with linear reference values
+            max_num_elements (int): max number of elements - 118 is a stretch
+        """
+        super().__init__()
+        self.register_buffer(
+            name="element_references",
+            tensor=element_references
+            if element_references is not None
+            else torch.zeros(max_num_elements + 1),
+        )
+
+    def _apply_refs(
+        self, target: torch.Tensor, batch: Batch, sign: int, reshaped: bool = True
+    ) -> torch.Tensor:
+        """Apply references batch-wise"""
+        indices = batch.atomic_numbers.to(
+            dtype=torch.int, device=self.element_references.device
```

Comment:
Is it possible that element_references ends up on a different device (CPU) and then everytime this method is called we copy over the incoming tensor (GPU) to CPU? 
If so maybe having a cached version of element_references for device might speed up and avoid extra inter device copies?

Reply:
Good point!  element references should always be on the device the trainer is on (see response to comment below). But I think I added this in here to make sure that the `batch.atomic_numbers` is always on the same device as the element references, because scatter throws an error otherwise.

I can't remember in what situation I was running into this, but let me investigate.